app/models/cart.rb

```diff
@@ -30,4 +30,15 @@ def total_price
   def remove(item_id)
     @contents.delete(item_id.to_s)
   end
+
+  def subtotal_of(item_id)
+    ((Item.find(item_id).price_per_unit * contents[item_id.to_s]).round(2))
+  end
+
+  def update_item(cart_params)
+    id = cart_params['item_id'].to_s
+    quantity = cart_params['quantity'].to_i
+    quantity = 0 if quantity < 0
```

Comment:
@tmikeschu Should we `remove(id) if quantity <= 0` so it doesn't appear in the cart anymore? As a shopper who wanted to delete an item I might prefer to see it totally gone.